### PR TITLE
add cloudbees-developers

### PR DESCRIPTION
**Pull request by car-roll**
# Description
This PR adds the team `cloudbees-developers` to plugins where CloudBees is the maintainer.
* [x] [antisamy-markup-formatter](https://github.com/jenkinsci/antisamy-markup-formatter-plugin) approved by @batmat 
* [x] [async-http-client](https://github.com/jenkinsci/async-http-client-plugin) approved by @batmat
* [x] [authentication-tokens](https://github.com/jenkinsci/authentication-tokens-plugin) approved by @batmat
* [x] [branch-api](https://github.com/jenkinsci/branch-api-plugin) (de facto) approved by @car-roll 
* [x] [config-file-provider](https://github.com/jenkinsci/config-file-provider-plugin) approved by @batmat
* [x] [pom-declarative-pipeline-migration-assistant-parent](https://github.com/jenkinsci/declarative-pipeline-migration-assistant-plugin/blob/master/pom.xml#L36) (de facto) approved by @car-roll 
* [x] [declarative-pipeline-migration-assistant-api](https://github.com/jenkinsci/declarative-pipeline-migration-assistant-plugin/tree/master/declarative-pipeline-migration-assistant-api) (de facto) approved by @car-roll 
* [x] [declarative-pipeline-migration-assistant](https://github.com/jenkinsci/declarative-pipeline-migration-assistant-plugin) (de facto) approved by @car-roll 
* [x] [display-url-api](https://github.com/jenkinsci/display-url-api-plugin) (de facto) approved by @car-roll 
* [x] [git](https://github.com/jenkinsci/git-plugin) approved by @MarkEWaite 
* [x] [git-client](https://github.com/jenkinsci/git-client-plugin) approved by @MarkEWaite 
* [x] [git-server](https://github.com/jenkinsci/git-server-plugin) approved by @batmat
* [x] [github-branch-source](https://github.com/jenkinsci/github-branch-source-plugin) (de facto) approved by @car-roll 
* [x] [jackson2-api](https://github.com/jenkinsci/jackson2-api-plugin) approved by @jtnord and @car-roll 
* [x] [msbuild](https://github.com/jenkinsci/msbuild-plugin) approved by @batmat
* [x] [scm-api](https://github.com/jenkinsci/scm-api-plugin) (de facto) approved by @car-roll 
* [x] [script-security](https://github.com/jenkinsci/script-security-plugin)  (de facto) approved by @car-roll 
* [x] [structs](https://github.com/jenkinsci/structs-plugin) (de facto) approved by @car-roll 
* [x] [variant](https://github.com/jenkinsci/variant-plugin) approved by @batmat
* [ ] [windows-slaves](https://github.com/jenkinsci/windows-slaves-plugin)
* [x] [mailer](https://github.com/jenkinsci/mailer-plugin)
* [x] [ldap](https://github.com/jenkinsci/ldap-plugin)

# Submitter checklist for adding or changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### For a newly hosted plugin only

- [ ] Add link to resolved HOSTING issue in description above

### For a new permissions file only

- [ ] Make sure the file is created in `permissions/` directory
- [ ] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [ ] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)
- [ ] Check that the file is named `plugin-${artifactId}.yml` for plugins

### When adding new uploaders (this includes newly created permissions files)

- [ ] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [ ] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [ ] Make sure to `@`mention the users being added so their GitHub account names are known if they require GitHub merge access (see below).
- [ ] [All newly added users have logged in to Artifactory at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### Reviewer checklist (not for requesters!)

- [ ] Check this if newly added person also needs to be given merge permission to the GitHub repo (please @ the people/person with their GitHub username in this issue as well). If needed, it can be done using an [IRC Bot command](https://jenkins.io/projects/infrastructure/ircbot/#github-repo-management)
- [ ] Check that the `$pluginId Developers` team has `Admin` permissions while granting the access.
- [ ] In the case of plugin adoption, ensure that the Jenkins Jira default assignee is either removed or changed to the new maintainer.
- [ ] If security contacts are changed (this includes add/remove), ping the security officer (currently `@Wadeck`) in this pull request. If an email contact is changed, wait for approval from the security officer.

There are [IRC Bot commands](https://jenkins.io/projects/infrastructure/ircbot/#issue-tracker-management) for it

### CODE REVIEW

File: permissions/plugin-declarative-pipeline-migration-assistant-api.yml
Comment (jtnord) [inline permissions/plugin-declarative-pipeline-migration-assistant-api.yml:9]: this looks incorrect YAML
```suggestion
  - "@cloudbees-developers"
```

File: permissions/plugin-declarative-pipeline-migration-assistant.yml
Comment (jtnord) [inline permissions/plugin-declarative-pipeline-migration-assistant.yml:9]: this looks incorrect YAML
```suggestion
  - "@cloudbees-developers"
```

File: permissions/plugin-jackson2-api.yml
Comment (jtnord) [inline permissions/plugin-jackson2-api.yml:11]: this plugin is using CD so there should be no need to grant this access - infact user level access should probably be revoked?

possibly something to do as a follow up for all plugins?

File: permissions/plugin-metrics.yml
Comment (jtnord) [inline permissions/plugin-metrics.yml:9]: already present on line 12/13 below :)

```suggestion
```

**Review by jtnord — COMMENTED** [review]
Some duplicates, and incorrect YAML.

That asside I approve where I am currently mentioned with deploy access for maintainership.

File: permissions/plugin-jackson2-api.yml
Comment (timja) [reply permissions/plugin-jackson2-api.yml:11]: up to plugins, generally people leave it in there in case there's issues and they need to release manually

File: permissions/plugin-declarative-pipeline-migration-assistant-api.yml
Comment (car-roll) [reply permissions/plugin-declarative-pipeline-migration-assistant-api.yml]: addressed in addd22728a8accbc679041d3116d0925cf72d248

File: permissions/plugin-declarative-pipeline-migration-assistant.yml
Comment (car-roll) [reply permissions/plugin-declarative-pipeline-migration-assistant.yml]: addressed in addd22728a8accbc679041d3116d0925cf72d248

File: permissions/plugin-metrics.yml
Comment (car-roll) [reply permissions/plugin-metrics.yml]: removed in addd22728a8accbc679041d3116d0925cf72d248

File: permissions/plugin-jackson2-api.yml
Comment (jglick) [reply permissions/plugin-jackson2-api.yml:11]: That is what we have done historically, especially when JEP-229 was brand new and we were still unsure whether it would work reliably. Usage on backports still remains to be validated AFAIK (https://github.com/jenkins-infra/jenkins-maven-cd-action/issues/4). Then on the other hand there are security arguments for revoking manual access to components with `cd.enabled`. A broader discussion that could be taken to the list I guess.

**Comment by jglick** [comment]
How would we see who is in this team? IIUC this is something in Artifactory rather than GitHub (https://github.com/orgs/jenkinsci/teams?query=cloudbees)?

File: permissions/pom-structs-parent.yml
Comment (jglick) [inline permissions/pom-structs-parent.yml:7]: Note that this component has been deleted in trunk: https://github.com/jenkinsci/structs-plugin/pull/114

**Comment by timja** [comment]
> How would we see who is in this team? IIUC this is something in Artifactory rather than GitHub (https://github.com/orgs/jenkinsci/teams?query=cloudbees)?

no it’s just a list that gets expanded in code in this repo, artifactory doesn’t know about it, list is defined here: https://github.com/jenkins-infra/repository-permissions-updater/blob/master/teams/cloudbees-developers.yml

in the code it just expands the reference and converts it to a set to remove duplicates

**Comment by jglick** [comment]
FTR https://github.com/jenkins-infra/repository-permissions-updater/blob/0c5cd6389079731bceda0babe00a4c8a50e78b9c/teams/cloudbees-developers.yml#L4-L19

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
It does not seem to be a right approach. Many of the plugins are not really "CloudBees maintained" anymore, e.g. Javadoc or JUnit. I suggest excluding all plugins with non-CloudBees maintainers from this PR

**Comment by jtnord** [comment]
> It does not seem to be a right approach. Many of the plugins are not really "CloudBees maintained" anymore, e.g. Javadoc or JUnit. I suggest excluding all plugins with non-CloudBees maintainers from this PR

@oleg-nenashev  that would remove everything as many people have since left CloudBees but are mentioned as maintainers.
I can not see the GitHub access on junit but the overwhelming majority in [plugins.jenkins.io](https://plugins.jenkins.io/junit/) says they where at one point employed by CloudBees.  
Rather than making a blanket blocking response  could you `-1` on the changes where you are a maintainer please, as the it is up to the maintainers of each plugin to agree or not.

**Comment by timja** [comment]
Last time this was done @batmat created a list with all the proposed plugins and checkboxes next to them with current maintainer's cc'ed.

That's a lot easier than this approach imo

**Comment by oleg-nenashev** [comment]
> Rather than making a blanket blocking response could you -1 on the changes where you are a maintainer please, as the it is up to the maintainers of each plugin to agree or not.

I am -1ing as the hosting team member. The current plugin governance is quite explicit about who is a maintainer. It is not tied to company affiliation

> that would remove everything as many people have since left CloudBees but are mentioned as maintainers

Yes. This is a technical debt accumulated by the company. See my proposals from 2018ish. It is not a job of the community to bypass the processes to reduce this technical debt

I agree with the previous approach by @batmat. It would at least get us somewhere gradually

**Comment by timja** [comment]
> I can not see the GitHub access on junit

You can see that here:

https://www.jenkins.io/doc/developer/publishing/source-code-hosting/

**Comment by timja** [comment]
RPU is the source of truth for release permission.

Which is a pain… @halkeye has a project he did some work on that would move GitHub to here as well to make it clearer

File: permissions/pom-structs-parent.yml
Comment (batmat) [reply permissions/pom-structs-parent.yml:7]: Worth removing then I believe, I'll do it.

**Review by batmat — APPROVED** [review]
I'm taking over this PR as Carroll is out on vacation.

I'm also approving this change for all plugins I'm maintainer of.

File: permissions/plugin-junit.yml
Comment (batmat) [inline permissions/plugin-junit.yml:10]: Reverting as per Oleg's comment, this way we can move and land the rest and keep discussing the more debatable plugins in a separate PR.

```suggestion
```

File: permissions/plugin-javadoc.yml
Comment (batmat) [inline permissions/plugin-javadoc.yml:9]: Reverting as per Oleg's comment, this way we can move and land the rest and keep discussing the more debatable plugins in a separate PR.

```suggestion
```

**Comment by batmat** [comment]
@oleg-nenashev I've removed junit & javadoc plugins to attend to your concerns. Any other plugin that you'd like to be removed? Thanks

**Comment by timja** [comment]
> @oleg-nenashev I've removed junit & javadoc plugins to attend to your concerns. Any other plugin that you'd like to be removed? Thanks

I'm the only one maintaining junit these days except for 1 release Jesse did while I was on holiday, if cloudbees wants to volunteer to help with it I have no issues but it depends on what the plan is...

**Comment by batmat** [comment]
@timja to be clear, 1) we have no evil hidden plans on suddenly changing all these plugins and break them and 2) we will make sure to keep abiding by the most active maintainers.

IOW, for the case of JUnit, our approach should be to still land a PR, and, for instance, if you Tim say you don't have the time to attend to it but the general idea seems fine, we would just take it to the finish line.

Of course, mistakes can happen, but if you see any behavior not akin to what I am describing above, please feel free to immediately reach out to me personally. I mean it.

**Comment by timja** [comment]
that's fine approved for Junit then

**Comment by MarkEWaite** [comment]
Approved for git client plugin and for git plugin.

**Comment by alecharp** [comment]
I approve for the mailer plugin

**Comment by basil** [comment]
[This YouTube video](https://www.youtube.com/watch?v=4Iem6JK6PtY&t=365s) illustrates why I am uncomfortable with the sentiment behind this PR. As summarized in [this blog post](https://blog.jessfraz.com/post/blurred-lines/#maintainership-for-a-project-must-be-earned):

> When an employee joins your company they should not automatically get push access to the project. I almost feel like I should repeat this because it is SO important to building trust with the community. **Everyone must play by the same rules.** EVERYONE.

The concept of people being given commit access by virtue of being part of a company does not seem simpatico with the concept of maintainership being earned. I have to agree with @oleg-nenashev in this regard.

One of the underlying assumptions behind this position is that maintainership is something one would desire to earn in the first place. I do not think people are exactly _lining up_ to become maintainers of the plugins mentioned in this PR. This raises a deeper question: how do we incentivize plugin maintenance within the community such that we don't need to put in place a corporate team to keep these plugins afloat?

I don't have the answers to these questions, and this comment is non-blocking. These are deep questions. I do not intend to debate with anyone, but I felt compelled to express my view. Please carry on without me.

**Comment by daniel-beck** [comment]
Not commenting on the merits of this specific change (philosophically I probably agree mostly with @basil, see e.g. `matrix-auth`), but it is consistent with the existing practices of adding maintainers, in that any existing maintainer can add/approve a new maintainer, and this has been done regularly even for relative newcomers AFAICT. In other words, this PR is fine given the list of approvals in the original comment. Whether that is maintained in a separate files for easy batch-updating, or put into each of these files individually, should not make a difference – as long as an existing team member approves a team addition.

**Comment by batmat** [comment]
I meant to post earlier, and time flew, again.

Basil, I've watched the talk back when you posted it. I see the point, and I would agree if this was here about Jenkins Core, which to me is more what compares to the Docker engine that Jessie discusses.
It would indeed never ever occur to me to grant permissions to Jenkins Core _upfront_ to newcomers, even if they're working in my company and I work with them on a daily basis. The Jenkins Project has always been behaving _very_ differently for plugins.

The point of this change is solely to make things slightly more manageable. As Daniel points out, we could already achieve the very same by filing dozens of PRs where one existing maintainer would approve and we'd add more folks everywhere. 

Because of all this, I'm still inclined to proceed with merging this change. 

I will obviously leave it a few hours or days to allow questioning my reasoning above, before I move forward with this merge.

**Comment by jglick** [comment]
I do not see any essential difference between core and plugins in this context, especially the foundational API plugins that form around half of this PR. If you do not feel comfortable handing out `jenkinsci/jenkins` commit rights to newcomers then you should not feel any more comfortable about doing so for, say, `jenkinsci/git-plugin` or (_shudder_) `jenkinsci/script-security-plugin`.

**Comment by jtnord** [comment]
windows-slaves was aleadred approved by me in https://github.com/jenkins-infra/repository-permissions-updater/pull/2265#pullrequestreview-837269904

**Comment by rsandell** [comment]
I approve ldap

**Comment by car-roll** [comment]
obsolete PR, closing

**Pull request closed without merging**
